docker/dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM ubuntu:18.04
+
+# Ubuntu setup
+ENV TZ=America/Chicago
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+# Setup environment variables for Docker image
+
+# setup home dir
+ENV HOME /root
+
+# Install dependencies from Debian package manager
+RUN apt-get update -y && \
+    apt-get upgrade -y && \
+    apt-get install -y software-properties-common build-essential && \
+    apt-get install -y wget git vim && \
+    add-apt-repository ppa:jonathonf/gcc-9.1 && \
+    apt-get install -y gcc-9 && \
+    apt-get install -y gfortran g++ autoconf libtool automake make cmake && \
+    apt-get install -y libglu1-mesa-dev freeglut3-dev mesa-common-dev libfreetype6-dev && \
+    apt-get install -y  libopenblas-dev liblapack-dev libarpack2-dev libarmadillo-dev
+
+# Install OCE-0.18.3
+RUN cd $HOME && \
+    wget https://github.com/tpaviot/oce/archive/OCE-0.18.3.tar.gz && \
+    tar -zxvf OCE-0.18.3.tar.gz && \
+    cd oce-OCE-0.18.3 && \
+    mkdir build && cd build  && \
+    cmake -DOCE_INSTALL_PREFIX:PATH=$HOME/opt/OCE-0.18.3 .. && \
+    make  -j4 && make install/strip
+RUN cd $HOME && \
+    rm ./OCE-0.18.3.tar.gz  && rm -rf ./oce-OCE-0.18.3
+## OCE evn
+RUN echo 'export OCE_DIR=$HOME/opt/OCE-0.18.3' >> $HOME/.bashrc
+RUN echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$OCE_DIR/lib' >> $HOME/.bashrc
+
+# Install CGM master from the repo
+RUN git clone https://bitbucket.org/fathomteam/cgm/src/master/ $HOME/opt/cgm && \
+    cd $HOME/opt/cgm && \
+    autoreconf -fi && \
+    mkdir build && cd build && \
+    ../configure --prefix=$HOME/opt/CGM --with-occ=$HOME/opt/OCE-0.18.3 --enable-shared && \
+    make && make install
+# CGM env
+RUN echo 'export PATH=$PATH:$HOME/opt/CGM/bin' >> $HOME/.bashrc
+RUN echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/opt/CGM/lib' >> $HOME/.bashrc
+RUN echo 'export C_INCLUDE_PATH=$C_INCLUDE_PATH:$HOME/opt/CGM/include' >> $HOME/.bashrc
+RUN echo 'export CPLUS_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:$HOME/opt/CGM/include' >> $HOME/.bashrc
```

Comment:
Same idea here, why not using `ENV` command, i.e. why this need to be in the `.bashrc` ?

Reply:
All these exports to `~/.bashrc` is not required in the install process. But it would be convenient for tests and running docker after installing. So, I put them here.

There are two `ENV` commands before installing `mcnp2cad`. These two are required and I used the `ENV`.

Reply:
yes but what I meant is those environment variable you export in the .bashrc, will only be available if you manually `source .bashrc` or run bash in your docker container.
I believe if you use ENV, they will be defined regardless of the way you use the container... (specially if you plan to use this in CI)

Reply:
Got it, thanks @bam241.

Reply:
![](https://d33wubrfki0l68.cloudfront.net/8d799fc311d166c3a7d9f1a7970e1dd1d43bf141/0a855/images/docker-env-vars/docker_environment_build_args.png)

from [[here](https://vsupalov.com/docker-arg-env-variable-guide/#arg-and-env-availability)]